### PR TITLE
Merge 1.0-dev into master

### PR DESCRIPTION
**Pull request by Kolaru**
As discussed, this would bring the work done in 1.0-dev into master.

I'd be happy if it could be reviewed, especially to make sure I did not drop any tests when merging.

Fix https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/85
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/94
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/127
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/182
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/209
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/229
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/255
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/316
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/317
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/319
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/345
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/346
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/410
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/416
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/434
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/464
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/470
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/474
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/491
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/495
#513 
https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/515
#538
#539

### CODE REVIEW

**Comment by OlivierHnt** [comment]
The CI is not passing because the new display mechanism does not work exactly the same for Julia 1.6.

For the time being, since the next Julia LTS may be out soon enough, what if we require Julia 1.8 on the CI? In which case we may also want to modify the combat entry in Project.toml.

**Comment by Kolaru** [comment]
Why does it work on Julia 1.0 though ? x)

I'm removing 1.0 and checking if Compat.jl easily solve our problem to keep 1.6 for now.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -12,7 +12,6 @@ jobs:
       matrix:
         version:
           - '1.6'
-          - '1'
```
Comment (lbenet) [inline .github/workflows/CI.yml:15]: I suggest to have back this (erased) line, so we can test also Julia 1.9 as well. (I may have been who removed it...)

File: Project.toml [hunk 2]
```diff
@@ -1,10 +1,12 @@
 name = "IntervalArithmetic"
 uuid = "d1acc4aa-44c8-5952-acd4-ba5d80a2a253"
 repo = "https://github.com/JuliaIntervals/IntervalArithmetic.jl.git"
-version = "0.20.8"
+version = "1.0.0"
```
Comment (lbenet) [inline Project.toml:4]: I think it was agreed to have this as 0.21.0 version...; see [this comment](https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/557#issuecomment-1588578133).

File: Project.toml [hunk 3]
```diff
@@ -15,21 +17,21 @@ SetRounding = "3cc68bcd-71a2-5612-b932-767ffbe40ab0"
...
+EnumX = "1"
 FastRounding = "0.2, 0.3"
-Polynomials = "0.7, 1, 2, 3"
+Polynomials = ">= 1.0"
 RecipesBase = "1.0"
 RoundingEmulator = "0.2"
 SetRounding = "0.2"
-StaticArrays = "0.8, 0.9, 0.10, 0.11, 0.12, 1.0"
-julia = "1.5"
+StaticArrays = "1.0"
```
Comment (lbenet) [inline Project.toml:28]: I think this constrains to versions of the form 1.0.x... Current version of `StaticArrays` is v1.6.0. A similar comment applies to `Polynomials`, and `CRlibm`.

**Comment by lbenet** [comment]
I've left few rather concrete comments (on the few files I could have a look). Question: Shall we still push changes to this branch, or wait until everything is reviewed?

File: Project.toml (hunk 3 above)
Comment (schillic) [reply Project.toml:28]: `1.0` is the same as `1` and stands for the interval [1.0.0, 2.0.0).
[Source](https://pkgdocs.julialang.org/v1/compatibility/#Caret-specifiers); note that the caret (^) is the default.

Comment (lbenet) [reply Project.toml:28]: Thanks for the clarification 👍

**Comment by OlivierHnt** [comment]
Fixes #85, #94, #127, #182, #209, #229, #255, #316, #317, #319, #345, #346, #410, #416, #434, #464, #470, #474, #491, #495, #515.

**Comment by Kolaru** [comment]
As discussed with @OlivierHnt this is now good, so I'm merging.

**Comment by Kolaru** [comment]
I then also went to close all related issues 🎉

That was a lot, thanks a lot to everyone who contributed :)

**Pull request merged**